addons/sale/models/sale.py

```diff
@@ -637,19 +637,23 @@ def _create_invoices(self, grouped=False, final=False, date=None):
             invoice_vals = order._prepare_invoice()
 
             # Invoice line values (keep only necessary sections).
+            number_of_notes_in_invoice_vals = 0
             for line in order.order_line:
                 if line.display_type == 'line_section':
                     pending_section = line
                     continue
-                if float_is_zero(line.qty_to_invoice, precision_digits=precision):
+                if line.display_type == 'line_note':
+                    number_of_notes_in_invoice_vals += 1
+                elif float_is_zero(line.qty_to_invoice, precision_digits=precision):
                     continue
-                if line.qty_to_invoice > 0 or (line.qty_to_invoice < 0 and final):
+                if line.qty_to_invoice > 0 or (line.qty_to_invoice < 0 and final) or line.display_type == 'line_note':
+                    line.sequence += len(invoice_vals_list) * 100
```

Comment:
Could be the opportunity to recompute a real sequence on all new lines instead of that strange computation here.

Reply:
👍 Implemented